Remove EC for sigops

ptschip opened the pull request:
This has never been used and introduced some small performance hits.

Remove sigops from CheckExcessive() and also did a bit of updating to the naming of some vars.

ptschip committed:
Add sigop counting back into ConnectBlock()

ptschip committed:
Update the sigops naming to MAX_BLOCK_SIGOPS_PER_MB and MAX_TX_SIGOPS_COUNT

ptschip committed:
Remove sigops from CheckExcessive()

ptschip committed:
Remove secondary loop for checking largest tx

Move checking the largest tx to the first loop in ContextualCheckBlock
and remove the second loop. There is no need to loop through the block
twice.

ptschip committed:
Remove excessive sigops test from excessive.py

ptschip committed:
fix format

gandrewstone commented on src/validation/validation.cpp:
```diff
@@ -2508,6 +2511,17 @@ bool ConnectBlockCanonicalOrdering(const CBlock &block,
+                    // this is to prevent a "rogue miner" from creating
+                    // an incredibly-expensive-to-validate block.
+                    nSigOps += GetP2SHSigOpCount(txref, view, flags);
+                    if (nSigOps > GetMaxBlockSigOpsCount(block.GetBlockSize()))
```
let's combine this check and the one above by putting them both outside the "if (!tx.IsCoinBase())" block (and for DTOR as well).

ptschip committed:
Use GetTransactionSigOpCount()

This aligns us more with the ABC code base and also makes
the code a bit cleaner.
